src/client/src/components/UserTimeline.jsx

```diff
@@ -220,7 +220,14 @@ const UserTimeline = () => {
       }
     });
   };
-  const notify = () => toast(t('New Job Alert Posted: ') + job.title);
+  const notify = () => {
+    if(job!=null){
+    toast(t('New Job Alert Posted: ') + job.title);
+    }
+    else{
+      toast(t('No jobs found based on your preferences'))
+    }
```

Comment:
just need to add this translation in i18next file

Reply:
from the code the rest seems good, i didnt test out the feature but i can see from the code it should work. i think u can merge it once you checked the translation

Reply:
added, i'll merge the PR now. Thanks!